### PR TITLE
Add an algorithm about Geolocation.

### PR DESCRIPTION
**Pull request by fuslonflare**
Please include a summary of the change. Please also include relevant motivation and context (if applicable).

- **Add an algorithm about Geolocation.**

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [ ] I have added comments to hard-to-understand areas of my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the comments
- [ ] I have made corresponding changes to the README.md

### CODE REVIEW

File: Algorithms/Other/GeoLocation.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Algorithms.Other
+{
+    public class GeoLocation
+    {
+        private static readonly double EarthRadius = 6371.01d;
```
Comment (siriak) [inline Algorithms/Other/GeoLocation.cs:9]: Codacy found an issue: [Replace this 'static readonly' declaration with 'const'.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6396283)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- Algorithms.Tests/Other/GeoLocationTests.cs  1
- Algorithms/Other/GeoLocation.cs  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6396283&bid=20741394)

File: Algorithms/Other/GeoLocation.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        /// <param name="lat2">Latitude of point B.</param>
+        /// <param name="lng2">Longitude of point B.</param>
+        /// <returns>Distance between point A to point B.</returns>
+        public static decimal CalculateDistanceFromLatLng(double lat1, double lng1, double lat2, double lng2)
+        {
+            var result = 0m;
+
+            try
+            {
+                if (lat1 == lat2 && lng1 == lng2)
```
Comment (siriak) [inline Algorithms/Other/GeoLocation.cs:25]: Codacy found an issue: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6396283)

File: Algorithms/Other/GeoLocation.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Algorithms.Other
+{
+    public class GeoLocation
```
Comment (siriak) [inline Algorithms/Other/GeoLocation.cs:7]: Codacy found an issue: [Add a 'protected' constructor or the 'static' keyword to the class declaration.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6396283)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- Algorithms.Tests/Other/GeoLocationTests.cs  1
- Algorithms/Other/GeoLocation.cs  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6396283&bid=20741394)

File: Algorithms/Other/GeoLocation.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        /// <param name="lat2">Latitude of point B.</param>
+        /// <param name="lng2">Longitude of point B.</param>
+        /// <returns>Distance between point A to point B.</returns>
+        public static decimal CalculateDistanceFromLatLng(double lat1, double lng1, double lat2, double lng2)
+        {
+            var result = 0m;
+
+            try
+            {
+                if (Math.Abs(lat1 - lat2) == 0d && Math.Abs(lng1 - lng2) == 0)
```
Comment (siriak) [inline Algorithms/Other/GeoLocation.cs:25]: Codacy found an issue: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6396283)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms.Tests/Other/GeoLocationTests.cs  1
- Algorithms/Other/GeoLocation.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6396283&bid=20741394)

File: C-Sharp.sln [hunk 5]
```diff
@@ -30,7 +30,7 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "DataStructures.Tests", "Dat
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Utilities", "Utilities\Utilities.csproj", "{3A41157D-296D-4BFC-A34E-91B5ED7F0905}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Utilities.Tests", "Utilities.Tests\Utilities.Tests.csproj", "{3A41157D-296D-4BFC-A34E-91B5ED7F0905}"
```
Comment (siriak) [inline C-Sharp.sln:33]: What was changed here?

**Review by siriak — CHANGES_REQUESTED** [review]

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms.Tests/Other/GeoLocationTests.cs  1
- Algorithms/Other/GeoLocation.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6396283&bid=20741394)

File: C-Sharp.sln (hunk 5 above)
Comment (fuslonflare) [reply C-Sharp.sln]: I forgot to exclude this file before commit, sorry. 😔🙇‍♂️
I commit the original value of solution file in [231e002](https://github.com/fuslonflare/C-Sharp/commit/231e002604a36e84e2195a48def8c9ac02b1db7e)

**Review by fuslonflare — COMMENTED** [review]
Thanks a lot for comment. it very useful
I resolved it in commit [231e002](https://github.com/fuslonflare/C-Sharp/commit/231e002604a36e84e2195a48def8c9ac02b1db7e).

**Review by siriak — APPROVED** [review]
Thanks!

**Pull request merged**